### PR TITLE
[CWS] make use of the remote workloadmeta in the system-probe

### PR DESCRIPTION
**Pull request by paulcacheux**
### What does this PR do?

This PR lays the foundation of being able to use the remote workloadmeta store in the system probe. This PR only allows the remote workloadmeta store to be used, since in most deployments the system probe cannot really maintain a working local store (cannot access the containerd socket, some configuration flags are only applied to the core agent etc).

Most of the changes from this PR are related to the transition from an optional workloadmeta component to a non-optional one.

The main motivations behind this change are:
- being able to use the tagger component (instead of the legacy constructor for which we are the last users)
- being able to drive the `containers_running` CWS telemetry in the system-probe instead of the security agent

### CODE REVIEW

**Review by spikat — APPROVED** [review]

File: tasks/security_agent.py [hunk 1]
```diff
@@ -381,6 +381,7 @@ def build_functional_tests(
     env["CGO_ENABLED"] = "1"
 
     build_tags = build_tags.split(",")
+    build_tags.append("test")
```
Comment (FlorentClarret) [inline tasks/security_agent.py:384]: Probably stupid question but: do you still need this one or was it a test?

**Review by FlorentClarret — APPROVED** [review]

File: tasks/security_agent.py (hunk 1 above)
Comment (paulcacheux) [reply tasks/security_agent.py:384]: It's actually needed for https://github.com/DataDog/datadog-agent/blob/b8ddf6c3b704dfca2ce0ecb381bb8509ab0826e9/comp/core/workloadmeta/impl/workloadmeta_mock.go#L6

**Review by akarpz — APPROVED** [review]

**Review by guyarb — APPROVED** [review]

**Review by ken-schneider — APPROVED** [review]

**Review by daniel-taf — APPROVED** [review]

**Comment by brycekahle** [comment]
Does this mean the workloadmeta component will be created, even if not used? Is it only CWS that requires this component?

**Comment by paulcacheux** [comment]
@brycekahle yes.. sadly it means that it will always be there. But since it's a dependency of the event monitor (not just CWS), I think it might be fine since it's enabled by default for other major system probe products. I would be happy to hear a solution to load it only if need honestly, I just didn't found one.

My testing on system-probe, but also on the security agent side where it's already in use shows that the impact at runtime is really low.

**Review by L3n41c — APPROVED** [review]
👍🏻 LGTM for files owned by @DataDog/container-integrations

**Review by sblumenthal — APPROVED** [review]

**Review by stanistan — APPROVED** [review]
LGTM for @DataDog/apm-onboarding

**Review by brycekahle — APPROVED** [review]

**Pull request merged**